release.py

```diff
@@ -0,0 +1,694 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Release automation script."""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+import github
+import github.GitRelease
+import github.Repository
+
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger('release')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+if 'GITHUB_TOKEN' not in os.environ:
+    raise SystemExit('Environment variable GITHUB_TOKEN not set.')
+
+auth = github.Auth.Token(os.environ['GITHUB_TOKEN'])
+gh_client = github.Github(auth=auth)
+
+
+VERSION_REGEX = r'(\d+\.\d+\.\d+(?:\.dev\d+)?)'
+VERSION_FILES = {
+    'ops/src': Path('ops/version.py'),
+    'ops/pyproject': Path('pyproject.toml'),
+    'testing': Path('testing/pyproject.toml'),
+    'tracing': Path('tracing/pyproject.toml'),
+    'uvlock': Path('uv.lock'),
+}
+
+
+def get_latest_release_tag(repo: github.Repository.Repository, branch_name: str) -> str | None:
+    """Get the latest release tag from the repository."""
+    releases = repo.get_releases()
```

Comment:
Are we certain this returns the releases in newest-first order?

Reply:
Yeah, if it's not guaranteed/specified in the docs, I think we should do a quick `.sort()` by `published_at`, to avoid surprises. Then again, they'll probably never change this ... it'd break too many people, so maybe it's okay.

Reply:
Sounds reasonable to keep it as is, thanks James.